### PR TITLE
Encrypt secrets

### PR DESCRIPTION
**Pull request by julien-nc**
* update composer deps
* update npm pkgs
* adjust to nc/vue 8
* switch to vite
* encrypt client_id, client_secret and all user's access/refresh tokens
* add a migration to encrypt existing stored values

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,10 +6,11 @@
...
     "test": "tests"
   },
+  "type": "module",
   "scripts": {
-    "build": "NODE_ENV=production webpack --progress --config webpack.js",
-    "dev": "NODE_ENV=development webpack --progress --config webpack.js",
-    "watch": "NODE_ENV=development webpack --progress --watch --config webpack.js",
+    "build": "NODE_ENV=production vite --mode production build",
+		"dev": "NODE_ENV=development vite --mode development build",
+		"watch": "NODE_ENV=development vite --mode development build --watch",
```
Comment (provokateurin) [inline package.json:13]: Wrong indentation

File: src/components/CreationForm.vue [hunk 2]
```diff
@@ -71,30 +71,6 @@
...
-					<template #singleLabel="{option}">
-						<component :is="option.icon"
-							v-if="option.icon"
-							class="multiselect-label-icon"
-							:size="20" />
-						<span class="option-title">
-							{{ option.label }}
-						</span>
-					</template>
-				</NcMultiselect>
```
Comment (provokateurin) [inline src/components/CreationForm.vue:97]: Why is this replaced without replacement?

File: lib/Controller/MiroAPIController.php [hunk 3]
```diff
@@ -108,6 +83,7 @@ public function getTeamAvatar(string $teamId, int $useFallback = 1) {
 	/**
 	 * @return DataResponse
 	 */
+	#[NoAdminRequired]
```
Comment (provokateurin) [inline lib/Controller/MiroAPIController.php:86]: So this was broken before?

File: lib/Settings/Admin.php [hunk 4]
```diff
@@ -35,8 +26,8 @@ public function getForm(): TemplateResponse {
 		$overrideLinkClick = $this->config->getAppValue(Application::APP_ID, 'override_link_click', '0') === '1';
 
 		$adminConfig = [
-			'client_id' => $clientID,
-			'client_secret' => $clientSecret,
+			'client_id' => $clientID === '' ? '' : 'dummy-client-id',
+			'client_secret' => $clientSecret === '' ? '' : 'dummy-client-secret',
```
Comment (provokateurin) [inline lib/Settings/Admin.php:30]: This change is not necessary, admins should be able to see which application is connected.

**Review by provokateurin — CHANGES_REQUESTED** [review]
Looks mostly good, just some small suggestions.
The cs:fixers went into the wrong commit though, please cleanup the history.

File: src/components/CreationForm.vue (hunk 2 above)
Comment (julien-nc) [reply src/components/CreationForm.vue:97]: Because it is not used.

File: lib/Settings/Admin.php (hunk 4 above)
Comment (julien-nc) [reply lib/Settings/Admin.php:30]: The inputs are type "password" so they won't see the value anyway.

File: lib/Controller/MiroAPIController.php (hunk 3 above)
Comment (julien-nc) [reply lib/Controller/MiroAPIController.php:86]: Yes I think so.

**Comment by julien-nc** [comment]
@provokateurin About the cs:fix in the wrong commit: That's fine. There is no development in this app anymore.

File: appinfo/info.xml [hunk 5]
```diff
@@ -4,7 +4,7 @@
 	<name>Miro integration</name>
 	<summary>Integration of Miro</summary>
 	<description><![CDATA[Miro integration lets users create boards, share board links and open boards directly in Nextcloud.]]></description>
-	<version>1.0.5</version>
+	<version>2.0.0</version>
```
Comment (provokateurin) [inline appinfo/info.xml:7]:
```suggestion
	<version>1.0.6</version>
```
Not a breaking change and no major new marketable feature, so just a patch bump is enough.

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Review by provokateurin — APPROVED** [review]

**Pull request merged**